### ISSUE TITLE
Support for redir in vroom tests

### ISSUE THREAD

**Issue by paulhybryant** [google/vroom#86]
Trying to test a syntax plugin.

Here is the vroom test:

### Test start
Vroom test for foldcol

Test environment setup.
  @system (RELAXED)
  > :set nocompatible<CR>
  > :set rtp+=$HOME/.vim/bundle/foldcol<CR>
  > :runtime! plugin/foldcol.vim<CR>
  > :set rtp+=$HOME/.vim/bundle/Align<CR>
  > :runtime! plugin/cecutil.vim<CR>
  > :runtime! plugin/AlignPlugin.vim<CR>
  > :runtime! plugin/AlignMapsPlugin.vim<CR>
  > :let b:folds = {}<CR>

Test plugin loaded
  > :echomsg g:loaded_foldcol<CR>
  ~ v3g

Try to fold column with default delimeter
  % 111111,222,33333<CR>
  % 1,222,333333333333333<CR>
  % 1111111111111,2,333
  > :FoldCol 2<CR>
  111111               , 222 , 33333
  1                         , 222 , 333333333333333
  1111111111111 , 2     , 333
  > :redir => b:synfold<CR>
  > :silent! syn list FoldCol2<CR>
  > :redir END<CR>
  > :echomsg "syn:".b:synfold<CR>
  ~ 1
### Test end

The previous test works fine. The last assert in the last statement is expected to fail "~ 1" in this case but it actually passes.

**Comment by dbarnett** [comment]
Could you define a test helper function in the plugin that uses `:redir` instead of trying to use it directly from vroom?

**Comment by paulhybryant** [comment]
Sure. That would work too.

**Comment by dbarnett** [comment]
We may still try to support `:redir` in some capacity, but FWIW it seems a little awkward in a vroom file anyway.

**Comment by paulhybryant** [comment]
Sure. One question though. With a helper function, I get the following error

Unexpected message:
FoldCol2       xxx match /\%16c...../  conceal

Expected message not received:
"FoldCol2       xxx match /\%16c...../  conceal" (verbatim mode)

The two message seems the same to me

Revised test below
    Vroom test for foldcol

    Test environment setup.
      @system (RELAXED)
      > :set nocompatible<CR>
      > :set rtp+=$HOME/.vim/bundle/foldcol<CR>
      > :runtime! plugin/foldcol.vim<CR>
      > :set rtp+=$HOME/.vim/bundle/Align<CR>
      > :runtime! plugin/cecutil.vim<CR>
      > :runtime! plugin/AlignPlugin.vim<CR>
      > :runtime! plugin/AlignMapsPlugin.vim<CR>
      > :let b:folds = {}<CR>


      Test plugin loaded
      > :echomsg g:loaded_foldcol<CR>
      ~ v3g


      Try to fold column with default delimeter
      % 111111,222,33333<CR>
      % 1,222,333333333333333<CR>
      % 1111111111111,2,333
      > :FoldCol 2<CR>
      111111        , 222 , 33333
      1             , 222 , 333333333333333
      1111111111111 , 2   , 333
      > :let b:syngroup = foldcol#GetSynGroup("FoldCol2")<CR>
      > :echomsg b:syngroup<CR>
      ~ FoldCol2       xxx match /\%16c...../  conceal

**Comment by dbarnett** [comment]
Yeah, looks the same to me. Any chance there's a tab character involved? You could also try weird things like slicing the string to see if it matches up to a certain point, or try base64 encoding it or something to verify the actual bytes.

**Comment by paulhybryant** [comment]
Thanks. Turns out there is a trailing space.

**Comment by dbarnett** [comment]
Another caveat: I checked your command and the output seems to vary according to window size even though it's not actually being printed to the screen (i.e., if the vim window is really narrow, there are more line wraps). It's unlikely in practice and probably best to just accept it will fail if your window is too narrow, but thought I'd mention it.

**Comment by paulhybryant** [comment]
Sounds good. Thanks for bringing that up!
